sdk/storage/azdatalake/file/client_test.go

```diff
@@ -472,9 +474,11 @@ func (s *UnrecordedTestSuite) TestCreateFileWithExpiryAbsolute() {
 	defer testcommon.DeleteFilesystem(context.Background(), _require, fsClient)
 
 	expiryTimeAbsolute := time.Now().Add(8 * time.Second)
-	expiry := file.CreationExpiryTypeAbsolute(expiryTimeAbsolute)
 	createFileOpts := &file.CreateOptions{
-		Expiry: expiry,
+		Expiry: file.CreateExpiryValues{
+			ExpiresOn:  time.Now().Add(8 * time.Second).UTC().Format(http.TimeFormat),
```

Comment:
Note that this does put the onus on the developer to get this correct.  We could add helper funcs to create a well-formed instance.  Maybe we do this later based on customer feedback.

Reply:
Yea, let's see customer feedback. I will include examples in this in my PR after we merge this PR.

Reply:
I would just reverse the initialization order because the mental model is:
1. Select an expiry type first and then
2. Set ExpiresOn (if necessary)